module/x/gravity/types/errors.go

```diff
@@ -5,15 +5,5 @@ import (
 )
 
 var (
-	ErrInternal                = sdkerrors.Register(ModuleName, 1, "internal")
-	ErrDuplicate               = sdkerrors.Register(ModuleName, 2, "duplicate")
-	ErrInvalid                 = sdkerrors.Register(ModuleName, 3, "invalid")
-	ErrTimeout                 = sdkerrors.Register(ModuleName, 4, "timeout")
-	ErrUnknown                 = sdkerrors.Register(ModuleName, 5, "unknown")
-	ErrEmpty                   = sdkerrors.Register(ModuleName, 6, "empty")
-	ErrOutdated                = sdkerrors.Register(ModuleName, 7, "outdated")
-	ErrUnsupported             = sdkerrors.Register(ModuleName, 8, "unsupported")
-	ErrNonContiguousEventNonce = sdkerrors.Register(ModuleName, 9, "non contiguous event nonce")
-	ErrInvalidGravityDenom     = sdkerrors.Register(ModuleName, 10, "invalid gravity denom")
-	ErrUnbonded                = sdkerrors.Register(ModuleName, 11, "signing validator not bonded")
+	ErrInvalid = sdkerrors.Register(ModuleName, 1, "invalid")
```

Comment:
this is a bad pattern to only have a single error, also number `1` is reserved for internal errors

Reply:
Most of the errors other than ErrInvalid were being used in incorrect ways, and the vast majority of errors in the codebase were already ErrInvalid. Adding correctly used error types to cover all of the many possible error with their own error types would be a big job, unsuitable for this cleanup PR.

Good catch on the number though. I will change it back to 3.